src/providers/google/youtube-favourite.ts

```diff
@@ -27,143 +35,173 @@ export default class YouTubeFavourite extends GoogleHandler {
-            // No results found, so stop sync
-            syncPosition.syncMessage = "Stopping. No results found.";
-            syncPosition = this.stopSync(syncPosition);
-    
-            return {
-                position: syncPosition,
-                results: [],
-            };
+
+        if (currentRange.startId) {
+            query.pageToken = currentRange.startId;
         }
 
-        const results = await this.buildResults(
-            serverResponse,
-            syncPosition.breakId,
+        const latestResponse = await youtube.videos.list(query);
+        const latestResult = await this.buildResults(
+            latestResponse,
+            currentRange.endId,
             _.has(this.config, "metadata.breakTimestamp")
                 ? this.config.metadata.breakTimestamp
                 : undefined
         );
-    
-        syncPosition = this.setNextPosition(syncPosition, serverResponse);
-    
-        if (results.length != this.config.batchSize) {
-            // Not a full page of results, so stop sync
-            syncPosition.syncMessage = `Processed ${results.length} items. Stopping. No more results.`;
-            syncPosition = this.stopSync(syncPosition);
-        }
-    
-        return {
-            results,
-            position: syncPosition,
-        };
-    }
-    
-    protected stopSync(syncPosition: SyncHandlerPosition): SyncHandlerPosition {
-        if (syncPosition.status == SyncHandlerStatus.ENABLED) {
-            return syncPosition;
+
+        items = latestResult.items;
+
+        let nextPageToken = _.has(latestResponse, "data.nextPageToken") ? latestResponse.data.nextPageToken : undefined;
+
+        if (items.length) {
+            rangeTracker.completedRange({
+                startId: items[0].sourceId,
+                endId: nextPageToken
+            }, latestResult.breakHit == SyncItemsBreak.ID);
+        } else {
+            rangeTracker.completedRange({
+                startId: undefined,
+                endId: undefined
+            }, false);
         }
 
-        syncPosition.status = SyncHandlerStatus.ENABLED;
-        syncPosition.thisRef = undefined;
-        syncPosition.breakId = syncPosition.futureBreakId;
-        syncPosition.futureBreakId = undefined;
+        if (items.length != this.config.batchSize) {
+            currentRange = rangeTracker.nextRange();
+            query = {
+                part: ["snippet", "contentDetails"],
+                myRating: "like",
+                maxResults: this.config.batchSize - items.length,
+            };
 
-        return syncPosition;
-    }
+            if (currentRange.startId) {
+                query.pageToken = currentRange.startId;
+            }
+
+            const backfillResponse = await youtube.videos.list(query);
+            const backfillResult = await this.buildResults(
+                backfillResponse,
+                currentRange.endId,
+                _.has(this.config, "metadata.breakTimestamp")
```

Comment:
Youtube doesn't support time ordering of activities.

Remove support for `breakTimestamp` in the handlers and the unit tests for all youtube.

Reply:
> Youtube doesn't support time ordering of activities.
> 
> Remove support for `breakTimestamp` in the handlers and the unit tests for all youtube.

Removed